pkg/api/controllers/volume.go

```diff
@@ -30,21 +30,23 @@ import (
 	c "github.com/opensds/opensds/pkg/context"
 	"github.com/opensds/opensds/pkg/controller/client"
 	"github.com/opensds/opensds/pkg/db"
+	dock "github.com/opensds/opensds/pkg/dock/client"
 	"github.com/opensds/opensds/pkg/model"
 	pb "github.com/opensds/opensds/pkg/model/proto"
 	. "github.com/opensds/opensds/pkg/utils/config"
 )
 
 func NewVolumePortal() *VolumePortal {
 	return &VolumePortal{
-		CtrClient: client.NewClient(),
+		CtrClient:  client.NewClient(),
+		DockClient: dock.NewClient(),
```

Comment:
I have a suggestion that you could remove DockClient field here, and just implement dock.NewClient() to CtrClient in VolumePortal. Something like this:
```go
func NewVolumePortal() *VolumePortal {
        ctrClient := func() client.Client {
                if installType == "thin" {
                        return dock.NewClient()
                }
                return client.NewClient()
        }
	return &VolumePortal{
		CtrClient: ctrClient,
        }
}
```

To make this goal, you need to update some methods in ProvisionDockClient interface to make it same with ControllerClient interface.